062-support-tiered-storage.md

```diff
@@ -0,0 +1,126 @@
+#  Support tiered storage natively in Strimzi
+
+## Current situation
+
+Kafka introduced tiered storage feature since 3.6. The feature is in early stage and it’s nice to support tiered storage configuration natively via Strimzi.
+
+## Motivation
+
+Supporting tiered storage via Strimzi allow Strimzi users to enable tiered storage natively with Strimzi config. By allowing to config custom configuration for `RemoteStorageManager`, users are able to adopt tiered storage feature on Strimzi easily.
+
+For simplicity, the proposal doesn't add support to customize `RemoteLogMetadataManager`. If tiered storage is enabled, the default implementation class `TopicBasedRemoteLogMetadataManager` in Kafka will be used.
+
+This proposal focuses on tiered storage support with a custom plugin brought in by the user. Other proposals in the future might add some built-in support for selected storage implementations such as S3. It should also be noted that, it’s on the end-user to ensure that their tiered storage setup works with Strimzi; the operator itself is strictly responsible for pushing down the necessary config to the broker pods.
+
+## Proposal
+
+Add a `spec.kafka.tieredStorage` field in the `Kafka` CR, which would support configuration for all tiered storage related configuration.
+
+The `tieredStorage` config contains configuration for some selected configuration of tiered storage feature. Most importantly, the configuration for `RemoteStorageManager` is needed.
+
+```
+kafka:
+  tieredStorage:
+    type: customRemoteStorageManager
+    className:
+    classPath:
+    config:
+    # A map with String keys and String values. Keys will be automatically prefixed with `rsm.config.`.
```

Comment:
`Keys will be automatically prefixed with `rsm.config.`` ... what does it mean? can you elaborate a little bit more? where and when they will be prefixed? It will be done by the operator when translating everything to a broker configuration but this comment here doesn't make it clear to the reader for the first time.

Reply:
Updated to `Keys will be automatically prefixed with rsm.config. and appended to Kafka broker config`

E.g. if I put a `storage.bucket.name: my-bucket` here, at Kafka broker config, a config called `rsm.config.storage.bucket.name: my-bucket` will be applied.

The translation should be appended by strimzi operator.

Reply:
Updated to `Keys will be automatically prefixed with rsm.config. and appended to Kafka broker config`

E.g. if I put a `storage.bucket.name: my-bucket` here, at Kafka broker config, a config called `rsm.config.storage.bucket.name: my-bucket` will be applied.

The translation should be appended by strimzi operator.